code/game/objects/items/inducer.dm

```diff
@@ -97,24 +97,37 @@
 /obj/item/inducer/proc/recharge(atom/movable/A, mob/user)
 	if(!isturf(A) && user.loc == A)
 		return FALSE
+	if (powertransfer>1000)
+		var/mob/living/carbon/human = user
+		if (human.electrocute_act( (powertransfer-1000)/400,human,stun = TRUE))
+			return FALSE
 	if(recharging)
 		return TRUE
 	else
 		recharging = TRUE
 	var/obj/item/stock_parts/cell/C = A.get_cell()
 	var/obj/O
 	var/coefficient = 1
-	if(istype(A, /obj/item/gun/energy))
+	if(istype(A, /obj/item/gun/energy) && !(obj_flags & EMAGGED))
 		to_chat(user,"Error unable to interface with device")
 		return FALSE
+	else
+		coefficient = 1/5
```

Comment:
This is going to set coefficient = 1/5 for everything

Reply:
fixed